crash when pokebag is full

```
2016-07-27 20:59:32,449 [    pgoapi] [ INFO] Nearby Pokemon: Magnemite, Growlithe
2016-07-27 20:59:32,855 [    pgoapi] [ INFO] Trying initiate catching Pokemon: Nickname: , Type: Magnemite, CP: 0, IV: 0.0
2016-07-27 20:59:33,162 [    pgoapi] [ INFO] Couldn't catch Nickname: , Type: NA, CP: 0, IV: 0.0 Your pokemon bag was full, attempting to clear and re-try
2016-07-27 20:59:33,983 [   pokecli] [ERROR] Error in main loop, restarting %s
Traceback (most recent call last):
  File "pokecli.py", line 121, in main
    api.main_loop()
  File "/home/sontek/venvs/pokemongo/src/pgoapi/pgoapi/pgoapi.py", line 848, in main_loop
    self.spin_all_forts_visible()
  File "/home/sontek/venvs/pokemongo/src/pgoapi/pgoapi/pgoapi.py", line 391, in spin_all_forts_visible
    map(lambda x: "via:%f,%f" % (x[0]['latitude'], x[0]['longitude']), destinations[1:]))
  File "/home/sontek/venvs/pokemongo/src/pgoapi/pgoapi/pgoapi.py", line 297, in walk_to
    self.spin_nearest_fort()
  File "/home/sontek/venvs/pokemongo/src/pgoapi/pgoapi/pgoapi.py", line 328, in spin_nearest_fort
    self.walk_to_fort(destinations[0], directly=True)
  File "/home/sontek/venvs/pokemongo/src/pgoapi/pgoapi/pgoapi.py", line 401, in walk_to_fort
    self.walk_to((fort['latitude'], fort['longitude']), directly=directly)
  File "/home/sontek/venvs/pokemongo/src/pgoapi/pgoapi/pgoapi.py", line 300, in walk_to
    while self.catch_near_pokemon() and catch_attempt <= self.max_catch_attempts:
  File "/home/sontek/venvs/pokemongo/src/pgoapi/pgoapi/pgoapi.py", line 445, in catch_near_pokemon
    catches_successful &= self.encounter_pokemon(target[0])
  File "/home/sontek/venvs/pokemongo/src/pgoapi/pgoapi/pgoapi.py", line 709, in encounter_pokemon
    return self.encounter_pokemon(pokemon, retry=True)
  File "/home/sontek/venvs/pokemongo/src/pgoapi/pgoapi/pgoapi.py", line 689, in encounter_pokemon
    encounter_id = pokemon_data['encounter_id']
AttributeError: Pokemon instance has no attribute '__getitem__'
2016-07-27 21:00:04,185 [   pokecli] [ INFO] Your given location:
```

--- a/pgoapi/pgoapi.py
+++ b/pgoapi/pgoapi.py
@@ -705,7 +705,7 @@ class PGoApi:
                 self.log.info("Couldn't catch %s Your pokemon bag was full, attempting to clear and re-try", pokemon)
                 self.cleanup_pokemon()
                 if not retry:
-                    return self.encounter_pokemon(pokemon, retry=True)
+                    return self.encounter_pokemon(pokemon_data, retry=True)
             else:
                 self.log.info("Could not start encounter for pokemon: %s", pokemon)
             return False

